src/plots/cartesian/layout_attributes.js

```diff
@@ -465,6 +465,36 @@ module.exports = {
             'Only has an effect if `anchor` is set to *free*.'
         ].join(' ')
     },
+    categorymode: {
+        valType: 'enumerated',
+        values: [
+            'trace', 'category ascending', 'category descending', 'array'
```

Comment:
for me just `ascending`\\`descending` doesn't have a clear relation to *alpha*numeric ordering. I definitely understand the aversion to multi-word strings, but I would vote for more description than `ascending`\\`descending`